### ISSUE TITLE
🧩 🐞 Bug Report: Conflict between Elementor Improved CSS Loading and CoBlocks 3.0.1

### ISSUE THREAD

**Issue by askwpgirl** [elementor/elementor#21802]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

When CoBlocks 3.0.1 is active on the site, Elementor > Settings > Features > Improved CSS Loading ACTIVE results in broken styles across multiple widgets. 

This issue does not occur with CoBlocks 2.25.5. The issue has been reported to CoBlocks as well. 

Elementor stylesheets also load in a different order when CoBlocks 3.0.1 plugin is active regardless of whether or not Improved CSS Loading is active or inactive. 

The result is catastrophic breaking of layouts and design for many widgets across the site. 

### Steps to reproduce

1. Elementor > Settings > Features > Improved CSS Loading set to ACTIVE
2. Edit a page with Elementor and add an icon box widget to the page.
3. Set the icon box setting to be Centered.
4. View the page.
5. The icon box widget will show its content centered. https://www.evernote.com/shard/s20/sh/4c0dd1f2-5e70-4db3-b473-3699601a8ea4/llfcNSWJI0WcCWl8BURpQpEMfhjlgEsOJCX_f0ORmoSWLmP1_LJRYX3qFQ
6. View the page source of the page. Take note of the order in which the Elementor stylesheets are loading. See screenshot: https://www.evernote.com/shard/s20/sh/633acd60-8de2-4c56-842e-c8989ac59851/tBZGw26WsYMzHjqZwzxTYxOA2hQ87NSl9jsl41DJ6M-kTVf3euu6HdNF-w
7. Activate CoBlocks 3.0.1 plugin.
8. Refresh the above page. The icon box items will no longer be centered. https://www.evernote.com/shard/s20/sh/16391c93-390d-43f2-b6e4-0f22e538b790/SmLtIAygPk87KEiHGoaz8VuvfDBOAzD8NCqQ9ZUYQaxpOH0Wr6D6YHAsgw
9. View the page source of the page. Take note of the order in which the Elementor stylesheets are loading. Notice, the are in a complete different order. See screenshot in step 6.
10. Revert CoBlocks to 2.25.5.
11. Refresh the above page. Note that the styling is correct.
12. View the page source. Note that the stylesheets are loading in the correct order as they should.

CoBlocks 3.0.1 appears to be changing the order in which Elementor stylesheets are loading but is also causing layouts to break if Improved CSS Asset Loading is active. This was not the case with the previous version of CoBlocks 2.25.5. Not sure where the responsibility lies here, but CoBlocks is very popular plugin used for formatting blog posts, so this problem shoud be addressed and made need cooperation between the two plugins. 

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [X] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [X] I can reproduce this bug consistently following the steps above.

### System Info

<details>
<summary>System Info</summary>

````txt

### wp-core ###

version: 6.1.1
site_language: en_US
user_language: en_US
timezone: Pacific/Honolulu
permalink: /%postname%/
https_status: true
multisite: false
user_registration: 0
blog_public: 1
default_comment_status: open
environment_type: staging
user_count: 1
dotorg_communication: true

### wp-paths-sizes ###

wordpress_path: /www
wordpress_size: 49.46 MB (51865201 bytes)
uploads_path: /www/wp-content/uploads
uploads_size: 6.01 MB (6303592 bytes)
themes_path: /www/wp-content/themes
themes_size: 16.95 MB (17774942 bytes)
plugins_path: /www/wp-content/plugins
plugins_size: 111.26 MB (116659799 bytes)
database_size: 7.69 MB (8060928 bytes)
total_size: 191.37 MB (200664462 bytes)

### wp-active-theme ###

name: Hello Elementor (hello-elementor)
version: 2.6.1 (latest version: 2.7.1)
author: Elementor Team
author_website: https://elementor.com/?utm_source=wp-themes&utm_campaign=author-uri&utm_medium=wp-dash
parent_theme: none
theme_features: core-block-patterns, widgets-block-editor, menus, post-thumbnails, automatic-feed-links, title-tag, html5, custom-logo, editor-style, align-wide, woocommerce, wc-product-gallery-zoom, wc-product-gallery-lightbox, wc-product-gallery-slider
theme_path: /www/wp-content/themes/hello-elementor
auto_update: Disabled

### wp-themes-inactive (5) ###

GeneratePress: version: 3.2.4, author: Tom Usborne (latest version: 3.3.0), Auto-updates disabled
Twenty Twenty: version: 2.0, author: the WordPress team (latest version: 2.1), Auto-updates disabled
Twenty Twenty-One: version: 1.6, author: the WordPress team (latest version: 1.7), Auto-updates disabled
Twenty Twenty-Three: version: 1.0, author: the WordPress team, Auto-updates disabled
Twenty Twenty-Two: version: 1.2, author: the WordPress team (latest version: 1.3), Auto-updates disabled

### wp-mu-plugins (1) ###

SiteDistrict Single Sign-on: version: 0.1.0, author: SiteDistrict Team

### wp-plugins-active (2) ###

CoBlocks: version: 3.0.1, author: GoDaddy, Auto-updates disabled
Elementor: version: 3.11.5, author: Elementor.com, Auto-updates disabled

### wp-plugins-inactive (9) ###

Akismet Anti-Spam: version: 5.0.2, author: Automattic (latest version: 5.1), Auto-updates disabled
Elementor Beta (Developer Edition): version: 1.1.1, author: Elementor.com, Auto-updates disabled
Elementor Pro: version: 3.9.2, author: Elementor.com, Auto-updates disabled
LearnDash Licensing & Management: version: 1.1.4, author: LearnDash, Auto-updates disabled
LearnDash LMS: version: 4.4.1.1, author: LearnDash, Auto-updates disabled
One Click Demo Import: version: 3.1.2, author: OCDI, Auto-updates disabled
Regenerate Thumbnails: version: 3.1.5, author: Alex Mills (Viper007Bond), Auto-updates disabled
WooCommerce: version: 7.1.0, author: Automattic (latest version: 7.5.1), Auto-updates disabled
WordPress Importer: version: 0.8, author: wordpressdotorg, Auto-updates disabled

### wp-media ###

image_editor: WP_Image_Editor_Imagick
imagick_module_version: 1687
imagemagick_version: ImageMagick 6.9.7-4 Q16 x86_64 20170114 http://www.imagemagick.org
imagick_version: 3.6.0
file_uploads: File uploads is turned off
post_max_size: 500M
upload_max_filesize: 500M
max_effective_size: 500 MB
max_file_uploads: 20
imagick_limits: 
	imagick::RESOURCETYPE_AREA: 122 MB
	imagick::RESOURCETYPE_DISK: 1073741824
	imagick::RESOURCETYPE_FILE: 786432
	imagick::RESOURCETYPE_MAP: 512 MB
	imagick::RESOURCETYPE_MEMORY: 256 MB
	imagick::RESOURCETYPE_THREAD: 1
imagemagick_file_formats: 3FR, AAI, AI, ART, ARW, AVI, AVS, BGR, BGRA, BGRO, BIE, BMP, BMP2, BMP3, BRF, CAL, CALS, CANVAS, CAPTION, CIN, CIP, CLIP, CMYK, CMYKA, CR2, CRW, CUR, CUT, DATA, DCM, DCR, DCX, DDS, DFONT, DNG, DPX, DXT1, DXT5, EPDF, EPI, EPS, EPS2, EPS3, EPSF, EPSI, EPT, EPT2, EPT3, ERF, FAX, FILE, FITS, FRACTAL, FTP, FTS, G3, G4, GIF, GIF87, GRADIENT, GRAY, GROUP4, H, HALD, HDR, HISTOGRAM, HRZ, HTM, HTML, HTTP, HTTPS, ICB, ICO, ICON, IIQ, INFO, INLINE, IPL, ISOBRL, ISOBRL6, J2C, J2K, JBG, JBIG, JNG, JNX, JP2, JPC, JPE, JPEG, JPG, JPM, JPS, JPT, JSON, K25, KDC, LABEL, M2V, M4V, MAC, MAGICK, MAP, MASK, MAT, MATTE, MEF, MIFF, MKV, MNG, MONO, MOV, MP4, MPC, MPEG, MPG, MRW, MSL, MTV, MVG, NEF, NRW, NULL, ORF, OTB, OTF, PAL, PALM, PAM, PATTERN, PBM, PCD, PCDS, PCL, PCT, PCX, PDB, PDF, PDFA, PEF, PES, PFA, PFB, PFM, PGM, PICON, PICT, PIX, PJPEG, PLASMA, PNG, PNG00, PNG24, PNG32, PNG48, PNG64, PNG8, PNM, PPM, PREVIEW, PS, PS2, PS3, PSB, PSD, PTIF, PWP, RADIAL-GRADIENT, RAF, RAS, RAW, RGB, RGBA, RGBO, RGF, RLA, RLE, RMF, RW2, SCR, SCT, SFW, SGI, SHTML, SIX, SIXEL, SPARSE-COLOR, SR2, SRF, STEGANO, SUN, TEXT, TGA, THUMBNAIL, TIFF, TIFF64, TILE, TIM, TTC, TTF, TXT, UBRL, UBRL6, UIL, UYVY, VDA, VICAR, VID, VIFF, VIPS, VST, WBMP, WMV, WPG, X, X3F, XBM, XC, XCF, XPM, XPS, XV, XWD, YCbCr, YCbCrA, YUV
gd_version: 2.2.5
gd_formats: GIF, JPEG, PNG, WebP, BMP, XPM
ghostscript_version: 9.26

### wp-server ###

server_architecture: Linux 5.4.0-1094-aws x86_64
httpd_software: nginx/1.18.0
php_version: 7.4.28 64bit
php_sapi: fpm-fcgi
max_input_variables: 5000
time_limit: 600s
memory_limit: 768M
max_input_time: 600s
upload_max_filesize: 500M
php_post_max_size: 500M
curl_version: 7.52.1 OpenSSL/1.0.2u
suhosin: false
imagick_availability: true
pretty_permalinks: true

### wp-database ###

extension: mysqli
server_version: 10.3.17-MariaDB-1:10.3.17+maria~bionic
client_version: mysqlnd 7.4.28
max_allowed_packet: 16777216
max_connections: 100

### wp-constants ###

WP_HOME: https://elementor.sitedistrict.com
WP_SITEURL: https://elementor.sitedistrict.com
WP_CONTENT_DIR: /www/wp-content
WP_PLUGIN_DIR: /www/wp-content/plugins
WP_MEMORY_LIMIT: 768M
WP_MAX_MEMORY_LIMIT: 768M
WP_DEBUG: false
WP_DEBUG_DISPLAY: true
WP_DEBUG_LOG: false
SCRIPT_DEBUG: false
WP_CACHE: false
CONCATENATE_SCRIPTS: undefined
COMPRESS_SCRIPTS: undefined
COMPRESS_CSS: undefined
WP_ENVIRONMENT_TYPE: staging
DB_CHARSET: utf8
DB_COLLATE: undefined

### wp-filesystem ###

wordpress: writable
wp-content: writable
uploads: writable
plugins: writable
themes: writable
mu-plugins: writable


````

</details>

**Comment by VerdiH** [comment]
Thank you for taking the effort to share this with us @askwpgirl 🙏

**Comment by askwpgirl** [comment]
@VerdiH - Is anyone at Elementor able to take responsibility and follow up with CoBlocks? I'm afraid this might be one of those situations where each plugin points at the other. This applies to all widgets.

**Comment by nicholaszein** [comment]
Hello Angela! 

Thank you for submitting your issue! 🙏 

Do you have a link to the issue reported to CoBlocks to include here as well? I believe, from what you're reporting, it seems they are doing something that affects how Elementor loads its styles. This would help understand what happened, and follow the progress of their fix.

Best regards

**Comment by nicholaszein** [comment]
Thank you for your feedback @AnthonyLedesma! 🙏 

We might get in touch with your team soon (if we haven't already). Verdi is the one reaching out to you so we can find a solution together.

I will pass this forward to the team so they can act on it!

**Comment by askwpgirl** [comment]
@nicholaszein and @AnthonyLedesma  - The reason that Nicholas has not replicated the issue is that I don't think he saw my reply follow up about the Improved CSS Loading option being enabled. However, even when this is not enabled, the order of the Elementor stylesheets is changed in the DOM either way. So please be sure to see set Elementor > Settings > Features > Improved CSS Loading set to ACTIVE to see the front-end issue of the visual changes to the widget styling changing thus catastrophically breaking sites across all widgets.

However, even with this off, you should be able to use View Page Source to see the order in which the stylesheets have been changed.

Please confirm you have checked these two things, so we can make faster progress with this. Neither of this things happened with CoBlocks 2.25.5. Here are Loom videos. Be sure to watch each one:

Change in order of stylesheets: https://www.loom.com/share/b13921bed03a45bfb097a341b5c2956e (this shows the DOM issue, i.e. the order of the stylesheets changing in the DOM)

Improved CSS Loading: https://www.loom.com/share/df1e3c670a4741e8afb7d10ed2666df2 (this shows the front-end issue when Improved CSS Loading is active)

So we have two issues:

1) The DOM issue wherein CoBlocks being active re-arranges the order in which Elementor's stylesheets load. This may or may not have visual impacts on the site depending on various site settings and page edits. It might have impacts on the site if one were to dig deeper, since stylesheet loading order is hyper critical. 

2) The front-end issue where widget styles break when Improved CSS Loading is active in Elementor Settings.

**Comment by askwpgirl** [comment]
I updated my above comment to indicate that the CoBlocks team may not have seen the issue due to not seeing follow up, not that Nicholas wasn't able to replicate.

**Comment by askwpgirl** [comment]
This looks like it was fixed in CoBlocks 3.0.2:

Fix issue around missing loader dependencies. [#2515](https://github.com/godaddy-wordpress/coblocks/pull/2515)

I tested with the update, and the Elementor issues are all resolved with this update. The order of stylesheets is now correct and the Improved CSS Loading now works fine as well.

**Comment by AnthonyLedesma** [comment]
@askwpgirl Thanks for that update! Glad to hear it was resolved in 3.0.2. We appreciate your efforts and your time in reporting this bug. Please do let us know if something happens again concerning CoBlocks!

**Comment by askwpgirl** [comment]
Thanks! Yes, since hopefully the bug report will help anyone on Elementor with CoBlocks 3.0.1 to know that updating to 3.0.2 or above will resolve the issue. Glad for quick fix.
